### ISSUE TITLE
matplotlib doesn't build with qhull-2020.2

### ISSUE THREAD

**Issue by HJarausch** [matplotlib/matplotlib#19243]
The library (and includes) (lib)qull have been deprecated in favor of the new *reentrant* version which is  (lib)qhull_r
On my Gentoo system I (only) have
/usr/lib64/libqhull_r.so

and

/usr/include/libqhull_r/*

Please fix matplotlib to work with this recent qhull

**Comment by tacaswell** [comment]
We appear to only use qhull from c and https://github.com/qhull/qhull/blob/master/src/libqhull/DEPRECATED.txt says 

> libqhull will be supported indefinitely."  

and http://www.qhull.org/html/qh-code.htm#library says

> Either version may be used. New code should use reentrant Qhull.

which seems a bit at-odds with the default behavior to not build the non-reentrant versions by default.  It looks like you can ask for the non-reentrant version of the libraries to be built (for reference it looks like Arch builds both versions) so I think that this should be fixed on the gentoo packaging side for now.

----

That said, we should still update the version of qhull we vendor and support (we currently are on 2015.2).

labeled as medium difficulty because it involves c code, but good first issue because the changes should be well isolated.

----

attn @ianthomas23 .  Will defer to Ian's judgement on the right path here.

**Comment by anntzer** [comment]
@tacaswell see https://github.com/matplotlib/matplotlib/pull/16720, I guess.

**Comment by ianthomas23** [comment]
Yes, we need to use the reentrant version of qhull. I submitted a PR to do this, and update to qhull 2019.1, about 10 months ago (#16571).

It stalled following discussion of whether we should continue to ship qhull code ourselves (my PR) or download it at build time (@anntzer's #16720). A decision is needed on this before we can progress.

**Comment by ianthomas23** [comment]
I have removed the "Good first issue" label as qhull upgrades invariably break some of our image tests, and dealing with these needs some judgement/experience.

**Comment by tacaswell** [comment]
Ah, great, glad we have options of how to fix this already open :)

**Comment by ianthomas23** [comment]
If we opt for downloading at build time (#16720) we'll need another PR to switch to downloading the latest qhull rather than 2015, and to change `qhull_wrap.c` to use the reentrant API.  The latter can mostly be cherry-picked from #16571.

**Comment by ianthomas23** [comment]
Closed by #16720.
